optas/builder.py

```diff
@@ -205,37 +207,136 @@ def get_model_states(self, name, time_deriv=0):
 
         """
         model = self.get_model(name)
-        assert time_deriv in model.time_derivs, f"model '{name}', was not specified with time derivative to order {time_deriv}"
+        assert (
+            time_deriv in model.time_derivs
+        ), f"model '{name}', was not specified with time derivative to order {time_deriv}"
         name = model.state_name(time_deriv)
         return self._decision_variables[name]
 
 
+    def get_model_parameters(self, name, time_deriv=0):
+        """Get the array of parameters for a given model.
+
+        Syntax
+        ------
+
+        parameters = builder.get_model_parameters(name, time_deriv=0)
+
+        Parameters
+        ----------
+
+        name (string)
+            Name of the model.
+
+        time_deriv (int)
+            The time-deriviative required (i.e. position is 0, velocity is 1, etc.)
+
+        Returns
+        -------
+
+        parameters (casadi.SX, with shape dim-by-T)
+            The array of parameters where dim is the number of model parameters, and T is the number of time-steps in the trajectory.
+
+        """
+        model = self.get_model(name)
+        assert time_deriv in model.time_derivs, f"model '{name}', was not specified with time derivative to order {time_deriv}"
+        name = model.parameter_name(time_deriv)
+        return self._parameters[name]
+
+
+    def get_model_parameter(self, name, t, time_deriv=0):
+        """Get the model parameter at a given time.
+
+        Syntax
+        ------
+
+        parameter = builder.get_model_parameter(name, t, time_deriv=0)
+
+        Parameters
+        ----------
+
+        name (string)
+            Name of the model.
+
+        t (int)
+            Index of the desired parameter.
+
+        time_deriv (int)
+            The time-deriviative required (i.e. position is 0, velocity is 1, etc.)
+
+        Returns
+        -------
+
+        parameter (casadi.SX, with shape dim-by-1)
+            The parameter vector where dim is the model number of parameters
+
+        """
+        parameters = self.get_model_parameters(name, time_deriv)
+        return parameters[:, t]
+
+
+    def get_robot_states_and_parameters(self, name, time_deriv=0):
```

Comment:
Man, you were the one who asked me to rename this to robot instead of model as i had in the beginning.
This is going in circles. Which one do you prefer?

Reply:
Did I? Sorry, I think because it hadn't occurred to me before that this functionality could be abstracted to the `TaskModel` class. Keep this as it is and we can change later. As I mention elsewhere, keep this PR for the `RobotModel` and create an issue - we can discuss there these sorts of changes.

Reply:
Added new issue.